File: Artist/app/src/main/java/com/example/artist/network/TopTracksService.java
package com.example.artist.network;


import com.example.artist.models.TopTracksResponse;
import com.example.artist.models.TopTracksResponseResult;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;


public interface TopTracksService {
    @GET("?method=track.search&format=json")
    Single<TopTracksResponseResult> getTopTracks(@Query("track") String user, @Query("api_key") String apiKey);

}

File: Artist/app/src/main/java/com/example/artist/network/TopTracksService.kt
package com.example.artist.network

import com.example.artist.models.TopTracksResponseResult
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface TopTracksService {
    @GET("?method=track.search&format=json")
    fun getTopTracks(@Query("track") user: String, @Query("api_key") apiKey: String): Single<TopTracksResponseResult>
}